lib/Dialect/Torch/Transforms/ReduceOpVariants.cpp

```diff
@@ -241,20 +268,32 @@ reduceNonValueTensorLiteralOpToValueTensorLiteralOp(NonValueTensorLiteralOp op,
 }
 
 namespace {
-class ReduceOpVariantsPass : public ReduceOpVariantsBase<ReduceOpVariantsPass> {
+struct ReduceOpVariantsPass
+    : public ReduceOpVariantsBase<ReduceOpVariantsPass> {
+  ReduceOpVariantsPass() = default;
+  ReduceOpVariantsPass(StringRef extraLibrary) {
+    this->extraLibrary = extraLibrary.str();
+  }
   void runOnOperation() override {
     MLIRContext *context = &getContext();
     RewritePatternSet patterns(context);
-    patterns.add<ConvertHasValueSemanticsOpsToValueTensors>(context);
+    OwningOpRef<ModuleOp> extraLibraryModule;
+    if (!extraLibrary.empty())
+      loadExtraLibrary(extraLibrary, context, extraLibraryModule);
+    patterns.add<ConvertHasValueSemanticsOpsToValueTensors>(context,
```

Comment:
I suggest creating a SymbolTable for the  extra library module and passing the symbol table to extraLibrary -- that will avoid a linear scan for each lookup.